tools/check_format.py

```diff
@@ -46,59 +48,49 @@
     "ProtobufUtil::MessageDifferencer": "Protobuf::util::MessageDifferencer"
 }
 
-found_error = False
-
-
-def printError(error):
-  global found_error
-  found_error = True
-  print "ERROR: %s" % (error)
-
 
 def checkNamespace(file_path):
   with open(file_path) as f:
     text = f.read()
     if not re.search('^\s*namespace\s+Envoy\s*{', text, re.MULTILINE) and \
        not 'NOLINT(namespace-envoy)' in text:
-      printError("Unable to find Envoy namespace or NOLINT(namespace-envoy) for file: %s" % file_path)
-      return False
-  return True
+      return ["Unable to find Envoy namespace or NOLINT(namespace-envoy) for file: %s" % file_path]
+  return []
 
 
 # To avoid breaking the Lyft import, we just check for path inclusion here.
 def whitelistedForProtobufDeps(file_path):
   return any(path_segment in file_path for path_segment in GOOGLE_PROTOBUF_WHITELIST)
 
-def findSubstringAndPrintError(pattern, file_path, error_message):
+def findSubstringAndReturnError(pattern, file_path, error_message):
   with open(file_path) as f:
     text = f.read()
     if pattern in text:
-      printError(file_path + ': ' + error_message)
+      error_messages = [file_path + ': ' + error_message]
```

Comment:
I assure you this is **not** the bottleneck. I did some profiling and 49% of the running time of the script is spent running clang-format (more precisely, waiting for the subprocess to finish) and 39% is running `tools/header_order.py` (which seems unreasonably expensive, but I suspect running it as a subprocess instead of inline incurs some overhead). The running time of this function is minuscule.

Reply:
OK -- I still don't like that this function is reading each file again, and then doing a line-by-line search.  But I believe you...it makes sense that clang is the real bottleneck.  Glad you found a relatively easy to way to get around that.  I'm not too familiar with the forking infrastructure in Python but I'll take a closer look.

Reply:
I'm going to leave this as is to keep this PR focused on just enabling parallelism.